docs/changed_pages.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import re
+
+import requests
+
+
+def fetch_changed_files(pr_num: str) -> list[str]:
+    url = f"https://api.github.com/repos/mlflow/mlflow/pulls/{pr_num}/files"
+    per_page = 100
+    changed_files = []
+    for page in range(1, 100):
+        r = requests.get(url, params={"page": page, "per_page": per_page})
+        r.raise_for_status()
+        files = r.json()
+        changed_files.extend(f["filename"] for f in files)
+        if len(files) < per_page:
+            return changed_files
+
+
+def main() -> None:
+    pr_num = os.environ.get("CIRCLE_PR_NUMBER")
+    if pr_num is None:
+        return
+
+    SOURCE_REGEX = re.compile(r"<!-- source: (.+) -->")
+    BUILD_DIR = pathlib.Path("build/html")
+    changed_files = fetch_changed_files(pr_num)
+    changed_pages: set[str] = set()
+    for p in BUILD_DIR.rglob("**/*.html"):
+        if m := SOURCE_REGEX.search(p.read_text()):
+            source = m.group(1)
+            if source in changed_files:
+                changed_pages.add(p.relative_to(BUILD_DIR))
```

Comment:
This only detects .rst file updates and doesn't detect .py file updates. Extra effort is required to support that.